docs/getting-started/list-filtering/sqlalchemy.rst

```diff
@@ -0,0 +1,283 @@
+==================
+SQLAlchemy Adapter
+==================
+
+The ``sqlalchemy_oso`` library can enforce policies over SQLAlchemy models.
+This allows policies to control access to collections of objects without
+needing to authorize each object individually.
+
+Usage
+=====
+
+.. todo:: More precise terminology (declarative models, etc.) in this
+   section.
+
+``sqlalchemy_oso`` works over your existing SQLAlchemy ORM models
+without modification.
+
+To get started, we need to:
+
+1. Make oso aware of our SQLAlchemy model types so that we can write policies
+   over them.
+2. Create a SQLAlchemy Session_ that uses oso to authorize access to data.
+
+**Register models with oso**
+
+We write a policy over our SQLAlchemy models.
+:py:func:`sqlalchemy_oso.register_models` registers all models
+that descend from a declarative base class as types that are available
+in the policy.
+
+Alternatively, the :py:meth:`oso.Oso.register_class` method can be called on
+each SQLAlchemy model that you will write rules for.
+
+**Create a SQLAlchemy Session that uses oso**
+
+oso performs authorization by integrating with SQLAlchemy sessions.
+Use the :py:func:`sqlalchemy_oso.authorized_sessionmaker` session
+factory instead of the default SQLAlchemy ``sessionmaker``. Every query made
+using sessions from this factory will have authorization applied.
+
+Before executing a query, oso consults the policy and obtains a list of
+conditions that must be met for a model to be authorized. These conditions
+are translated into SQLAlchemy expressions and applied to the query before
+retrieving objects from the database.
+
+.. _Session: https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session
+
+Example
+=======
+
+Let's look at an example usage of this library.  Our example is a social media
+app that allows users to create posts.  There is a ``Post`` model and a ``User``
+model:
+
+.. literalinclude:: /examples/list-filtering/sqlalchemy/sqlalchemy_example/models.py
+    :caption: :fab:`python` models.py
+    :language: python
+
+
+Now, we'll write a policy over these models.  Our policy contains the following
+rules:
+
+1. A user can read any public post.
+2. A user can read their own private posts.
+3. A user can read private posts for users they manage (defined through the
+   ``user.manages`` relationship).
+4. A user can read all other users.
+
+.. literalinclude:: /examples/list-filtering/sqlalchemy/sqlalchemy_example/policy.polar
+    :caption: :fa:`oso` policy.polar
+    :language: polar
+
+.. note::
+
+    The SQLAlchemy integration is deny by default.  The final rule for ``User``
+    is needed to allow access to user objects for any user.
+
+    If a query is made for a model that does not have an explicit rule in the
+    policy, no results will be returned.
```

Comment:
As a follow-up it would be good to try to make this clearer. Especially the part that we're talking about sub-queries for authorization data (e.g., we're looking up `User` data as part of determining authorization for `Post`)

Reply:
agreed